core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -365,15 +365,35 @@ default boolean hasOwnAttr(AttributeKey<?> key) {
     MeterRegistry meterRegistry();
 
     /**
-     * Triggers the current timeout immediately if a timeout was scheduled before.
-     * Otherwise, the current {@link Request} will be timed-out immediately after a timeout scheduler is
-     * initialized.
+     * Cancels the current {@link Request} with a {@link Throwable}.
+     */
+    void cancel(Throwable cause);
+
+    /**
+     * Shortcut for {@code cancel(RequestCancellationException.get())}.
+     */
+    void cancel();
+
+    /**
+     * Shortcut for {@code cancel(RequestTimeoutException.get())} or
+     * {@code cancel(ResponseTimeoutException.get())}.
      */
     void timeoutNow();
```

Comment:
How about overriding this method in `ClientRequestContext` and `ServiceRequestContext` to provide better Javadoc there? e.g.

```java
/**
 * Shortcut for {@code cancel(RequestTimeoutException.get()}.
 */
@Override
void timeoutNow();
```